lyricsources/subtitles4songs/subtitles4songs.py

```diff
@@ -0,0 +1,109 @@
+S4S_SEARCH_PATH = '/en/subtitles/subtitles4songs.aspx'
+S4S_SUBTITLE_PATH = '/en/subtitles/getsubtitle.aspx'
+S4S_SEARCH_RESULT_PATTERN = re.compile(r'<a [^<]*?href="getsubtitle\.aspx(\?artist=.*?song=.*?)">', re.DOTALL)
+S4S_LRC_PATTERN = re.compile(r'<span id="ctl00_ContentPlaceHolder1_lbllyrics"><h3>(.*?)<\/h3>(.*?)<\/span>', re.DOTALL)
+TITLE_PATTERN = re.compile(r'&song=(.*$)')
+ARTIST_PATTERN = re.compile(r'artist=(.*?)&')
+
+gettext.bindtextdomain('osdlyrics')
+gettext.textdomain('osdlyrics')
+
+
+class Subtitles4songsSource(BaseLyricSourcePlugin):
+    """ Lyric source from https://www.rentanadviser.com/en/subtitles/subtitles4songs.aspx
+    """
+
+    def __init__(self):
+        super().__init__(id='subtitles4songs', name=_('subtitles4songs'))
+
+    def do_search(self, metadata):
+        # type: (osdlyrics.metadata.Metadata) -> List[SearchResult]
+        keys = []
+        if metadata.artist:
+            keys.append(metadata.artist)
+        if metadata.title:
+            keys.append(metadata.title)
+        urlkey = '+'.join(keys).replace(' ', '+')
+        url = S4S_HOST + S4S_SEARCH_PATH
+        print("urlkey:"+urlkey)
+        print("url:"+url)
+        status, content = http_download(url=url,
+                                        params={'q': urlkey},
+                                        proxy=get_proxy_settings(self.config_proxy))
+        if status < 200 or status >= 400:
+            raise http.client.HTTPException(status, '')
+        print(status); print(content);
+        match = S4S_SEARCH_RESULT_PATTERN.findall(content.decode('utf8'))
+        print(match)
+        result = []
+        if match:
+            for match_elem in match:
+                title = unquote(TITLE_PATTERN.search(match_elem).group(1))
+                artist = unquote(ARTIST_PATTERN.search(match_elem).group(1))
+                url = S4S_HOST + S4S_SUBTITLE_PATH + match_elem + "&type=lrc"
+                print(title)
+                print(artist)
+                print(url)
+                if url is not None:
+                    result.append(SearchResult(title=title,
+                                               artist=artist,
+                                               sourceid=self.id,
+                                               downloadinfo=url))
+        return result
+
+    def do_download(self, downloadinfo):
+        print("downloadinfo"); print(downloadinfo);
+        # type: (Any) -> bytes
+        status, content = http_download(downloadinfo,
+                                        proxy=get_proxy_settings(self.config_proxy))
+        if status < 200 or status >= 400:
+            raise http.client.HTTPException(status)
+        if content:
+            print('content:'); print(content)
+            content = S4S_LRC_PATTERN.search(content.decode('utf8')).group(2)
+            # Replacing html breaks with new lines.
+            content = re.sub('<br />', "\n", content)
```

Comment:
:thinking:  I didn't found any of those in the results I've inspected from `getsubtitle.aspx`, do you have an example?

Reply:
I think every single one of them has `<br />` tags.

Try `view-source:https://www.rentanadviser.com/en/subtitles/getsubtitle.aspx?artist=David%20Guetta&song=Titanium%20ft.%20Sia` (remember to look at the source code and not the rendered page)

Reply:
Aaaaaaa, you're crawling the website, while I was trying downloading it with a session:

```
curl 'https://www.rentanadviser.com/en/subtitles/getsubtitle.aspx?artist=David+Guetta&song=Titanium+ft.+Sia&type=lrc' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:90.0) Gecko/20100101 Firefox/90.0' -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8' -H 'Accept-Language: en-US,en;q=0.8,pt-BR;q=0.5,es;q=0.3' --compressed -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: https://www.rentanadviser.com' -H 'Connection: keep-alive' -H 'Referer: https://www.rentanadviser.com/en/subtitles/getsubtitle.aspx?artist=David%20Guetta&song=Titanium%20ft.%20Sia&type=lrc' -H 'Cookie: ASP.NET_SessionId=v1x3q5rp3hpu4g1jr25g3uvk' -H 'Upgrade-Insecure-Requests: 1' -H 'Sec-Fetch-Dest: document' -H 'Sec-Fetch-Mode: navigate' -H 'Sec-Fetch-Site: same-origin' -H 'Sec-Fetch-User: ?1' -H 'Pragma: no-cache' -H 'Cache-Control: no-cache' -H 'TE: Trailers' --data-raw '__EVENTTARGET=ctl00%24ContentPlaceHolder1%24btnlyrics&__EVENTARGUMENT=&ctl00%24ContentPlaceHolder1%24txtsearchsubtitle=Search+Lyrics+%26+Subtitles&ctl00%24ContentPlaceHolder1%24Overcome_Enter_problem_in_IE2=&__VIEWSTATEGENERATOR=2245B277&__EVENTVALIDATION=%2FwEdAAaHOmx%2Bh7BWQCNEhvgO%2BUE2RuJmd8xHp1FvBUnaqp%2B06DOW6IPr5umq7%2FycngDpTXM1RCHQqe6x74z2AXzs4B32JjCeau28td%2F8t%2FqvcxlGDE8RmFoVYBUtr01iOVUfHho2tBGalsBI5fy71AiczchqRqMTxNmjo8fByi56acZTOg%3D%3D&__VIEWSTATE=%2FwEPDwUKLTUxMDQ1NjY2NQ8WBh4JUm9vdGhQYXRoBUNDOlxJbmV0cHViXHZob3N0c1xyZW50YW5hZHZpc2VyLmNvbVxodHRwZG9jc1xlblxBcHBfRGF0YVxzdWJzXGRhdGFcHgdzdWJQYXRoBWRDOlxJbmV0cHViXHZob3N0c1xyZW50YW5hZHZpc2VyLmNvbVxodHRwZG9jc1xlblxBcHBfRGF0YVxzdWJzXGRhdGFcRGF2aWQtR3VldHRhLVRpdGFuaXVtLWZ0Li1TaWEuc3J0HglQYWdlUGFybXMFOT9hcnRpc3Q9RGF2aWQlMjBHdWV0dGEmc29uZz1UaXRhbml1bSUyMGZ0LiUyMFNpYSZ0eXBlPWxyY2RkldCkUb4gqLLarJTey5JNJCWNbyGG7G%2FEUyKfIaQ4lZ8%3D'
```